libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TripsManagerImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.text.format.DateUtils;
+import androidx.annotation.NonNull;
+import com.mapbox.android.accounts.v1.MapboxAccounts;
+
+class TripsManagerImpl implements TripsManager {
+
+  enum RotateTripSku {
+    INVALID,
+    ROTATE_ON_TIMER_EXPIRE,
+    ROTATE_ON_REQUEST_COUNT_EXPIRE
+  }
+
+  private static final String MAPBOX_NAV_PREFERENCE_TRIPS_SKU = "com.mapbox.navigationsdk.accounts.trips.sku";
+  private static final String MAPBOX_NAV_PREFERENCE_ROUTE_REQ_COUNT = "com.mapbox.navigationsdk.accounts.trips.count";
+  private static final String MAPBOX_NAV_PREFERENCE_TRIPS_TIMESTAMP = "com.mapbox.navigationsdk.accounts.trips.time";
+  private static final int ROUTE_REQUEST_COUNT_THRESHOLD = 5;
+  private static final int TRIPS_TIMER_EXPIRE_THRESHOLD = 2;
+  private static final long TRIPS_TIMER_EXPIRE_AFTER = (DateUtils.HOUR_IN_MILLIS / 1000) * TRIPS_TIMER_EXPIRE_THRESHOLD;
+  private RotateTripSku rotateSku = RotateTripSku.INVALID;
+  private AccountsPreference accountsPreference;
+
+  TripsManagerImpl(AccountsPreference accountsPreference) {
+    this.accountsPreference = accountsPreference;
+  }
+
+  private void refreshSkuToken() {
+    if (!shouldRefreshSku()) {
+      return;
+    }
+    int requestCount;
+    switch (rotateSku) {
+      case ROTATE_ON_TIMER_EXPIRE:
+        requestCount = getRouteRequestCountThreshold();
+        requestCount++;
+        setTimerExpiry();
+        persistTripsSkuToken();
+        break;
+      case ROTATE_ON_REQUEST_COUNT_EXPIRE:
+        requestCount = 0;
+        setTimerExpiry();
+        persistTripsSkuToken();
+        break;
+      default:
+        requestCount = getRouteRequestCountThreshold();
+        requestCount++;
+        break;
+    }
+    setRouteRequestCountThreshold(requestCount);
+  }
+
+  private boolean shouldRefreshSku() {
+    boolean routeReqCountExpired = validateRouteRequestCountExpiry();
+    boolean timerExpired = validateTimerExpiry();
+    if (routeReqCountExpired) {
+      rotateSku = RotateTripSku.ROTATE_ON_REQUEST_COUNT_EXPIRE;
+    } else if (timerExpired) {
+      rotateSku = RotateTripSku.ROTATE_ON_TIMER_EXPIRE;
+    } else {
+      rotateSku = RotateTripSku.INVALID;
+    }
+    return routeReqCountExpired || timerExpired;
+  }
+
+  private boolean validateTimerExpiry() {
+    long skuTokenTimeStamp = getTimerExpiry();
+    return isTwoHoursExpired(skuTokenTimeStamp);
+  }
+
+  private boolean validateRouteRequestCountExpiry() {
+    int routeRequestCount = getRouteRequestCountThreshold();
+    return routeRequestCount > ROUTE_REQUEST_COUNT_THRESHOLD;
+  }
+
+  private void setRouteRequestCountThreshold(int count) {
+    accountsPreference.getPreferenceManager().set(MAPBOX_NAV_PREFERENCE_ROUTE_REQ_COUNT, count);
```

Comment:
what's benefit generic `AccountsPreference` wrapper is adding here? wouldn't it incur an overhead of double conversion from primitive to object and back?